htlcswitch/failure_detail.go

```diff
@@ -1,55 +1,55 @@
 package htlcswitch
 
-// FailureDetail is an enum which is used to enrich failures with
-// additional information.
-type FailureDetail int
+// OutgoingFailure is an enum which is used to enrich failures which occur in
+// the switch or on our outgoing link with additional metadata.
+type OutgoingFailure int
 
 const (
-	// FailureDetailNone is returned when the wire message contains
+	// OutgoingFailureNone is returned when the wire message contains
 	// sufficient information.
-	FailureDetailNone = iota
+	OutgoingFailureNone OutgoingFailure = iota
 
-	// FailureDetailOnionDecode indicates that we could not decode an
+	// OutgoingFailureOnionDecode indicates that we could not decode an
 	// onion error.
-	FailureDetailOnionDecode
+	OutgoingFailureOnionDecode
```

Comment:
Why is onion decode an outgoing error? This commit is just are name, but maybe it could use a comment.

Reply:
This is a bit of a niche one, it happens when one of our own payments fail on our outgoing link, but the router cannot decode the `lnwire.FailureMessgae` (I've renamed to reflect this). It's an error that occurred for an outgoing error (and shouldn't occur), didn't seem like it was worth adding a new error type.